### PR TITLE
added bonus assignments to grading

### PR DESCRIPTION
**Pull request by bjorn-out**
It's a bit repeat-y, but this seems to be the only way without making changes to the course website.

In order to keep the file more readable, I haven't put comments in there. Instead, I'll explain the calculation here:

```yaml
automatic:
    correctness: (check_score * 5).floor
    bonus: check_score.floor * 2
calculation: (bonus / 30 * 9 * 6).round(1)
```

- The 2 is the amount of points which can be earned for this bonus assignment.
- 30 is the amount of points which can normally be earned in total for this course.
- The 9 comes from the formula for calculating a grade: `points / available_points * 9 + 1`
- 6 is the total weight of the grades for the regular modules.

In short: `bonus / 30 * 9` is the amount your grade will increase (or 0 if check50 fails). We multiply this by 6 because the weight `bonus` in the `mods` calculation just adds whatever grade the assignment has to the scores total, but doesn't change the total weight. This means it will be divided by the total weight of the weighted modules on calculation of the final grade.

### CODE REVIEW

**Comment by bjorn-out** [comment]
Conflict resolved.
